widgetastic_manageiq/__init__.py

```diff
@@ -674,59 +674,68 @@ def read(self):
         return [cb for cb in self.checkboxes if cb.selected]
 
 
-class BootstrapSwitchSelect(CheckboxSelect):
+class BootstrapSwitchSelect(View):
     """BootstrapSwitchSelect view.
 
-    This view is very similar to parent CheckboxSelect view. BootstrapSwitches used instead of
-    usual Checkboxes. It can be found in the same policy's events assignment screen since
-    CFME 5.8.1.
+    This view is very similar to CheckboxSelect view. BootstrapSwitches used instead of
+    usual Checkboxes. It can be found in policy's events assignment screen since CFME 5.8.1.
 
     """
 
-    BS_TEXT = '/../../following-sibling::text()[1]'
+    ROOT = ParametrizedLocator(".//div[@id={@search_root|quote}]")
+    BS_LOCATOR = ".//text()[normalize-space(.)='{}']/preceding-sibling::div[1]//input"
 
-    def _get_bs_description(self, bs):
-        """Returns text description of the BootstrapSwitch widget.
+    def __init__(self, parent, search_root, logger=None):
+        View.__init__(self, parent, logger=logger)
+        self.search_root = search_root
 
-        We have to use such hack with the script execution, because Selenium cannot return text of a
-        text node itself.
+    def _get_attr_from_text(self, attr, text):
+        return self.browser.element(self.BS_LOCATOR.format(text)).get_attribute(attr)
 
-        Returns: str
+    @cached_property
+    def all_labels(self):
+        elements = self.browser.elements(".//div[@class='form-horizontal']")
+        labels = []
+        for el in elements:
+            labels.extend(self.browser.text(el).split(" Yes No ")[1:])
```

Comment:
Man, this is sooo crude :D

Reply:
I didn't find better way to get these strings 👅